pyanaconda/ui/webui/__init__.py

```diff
@@ -90,25 +107,62 @@ def run(self):
         """Run the interface."""
         log.debug("web-ui: starting cockpit web view")
 
-        # Force Firefox to be used via the BROWSER environment variable.
-        # This is read by cockpit-desktop and makes it launch Firefox in kiosk mode
-        # instead of the GTK WebKit based web view it launches by default.
+        try:
+            if conf.system.provides_liveuser:
+                self._watch_webui_on_live()
+            else:
+                self._run_webui()
+        finally:
+            self._remove_initialized_backend_flag()
```

Comment:
I think you should create and remove the file in the same function. It will be easier to follow.

Reply:
Resolved by context manager. I was thinking about this before and didn't done that only because I wanted to have it as much optimized as possible. However, I changed it because between the `setup` and `run` calls are just simple calls which shouldn't take any significant time and this seems to be much easier to understand.